File: 3. Linked List/8_is_univalue_list.java
class Node<T> {
    T val;
    Node<T> next;

    public Node(T val) {
        this.val = val;
        this.next = null;
    }
}

class Source {
    public static <T> boolean isUnivalueList(Node<T> head) {

        Node<T> current = head;
        T value = current.val;

        while (current != null) {
            if (current.val != value) {
                return false;
            }
            current = current.next;
        }

        return true;
    }
}

File: 3. Linked List/8_is_univalue_list.py
class Node:
    def __init__(self, val):
        self.val = val
        self.next = None

def is_univalue_list(head):
    current = head
    value = current.val

    while current is not None:
        if current.val != value:
            return False
        current = current.next

    return True
